site/docs/contribute.md

```diff
@@ -45,6 +45,16 @@ The Iceberg community prefers to receive contributions as [Github pull requests]
 * If a PR is related to an issue, adding `Closes #1234` in the PR description will automatically close the issue and helps keep the project clean
 * If a PR is posted for visibility and isn't necessarily ready for review or merging, be sure to convert the PR to a draft
 
+### Merging Pull Requests
+
+Most pull requests can be merged once a single committer is satisfied with the code in the PR. Before merging all review comments should be addressed either by making changes or agreeing the request is out of scope for the PR. For larger changes or additions to public APIs committers will wait at least 24 hours before merging to ensure there is no additional feedback from members of the community. As in all ASF governed projects committers are expected to act in the best interest of the project. If a committer feels there might be a conflict of interest with a pull request they review, they are encouraged to ask for another committer to review the pull request.
+
+Requesting changes on a PR indicates a reviewer believes the PR has merit but still needs issues addressed before merging. If a reviewer believes the change should not be merged at all and there is nothing the author could do to address the reviewers concerns (e.g. the reviewer feels the change should go through the [Iceberg improvement proposal](#apache-iceberg-improvement-proposals)), the reviewer should explicitly state this on the PR. In the rare event that a PR author and reviewers cannot come to a consensus on a PR, the disagreement should be raised to the developer mailing list for further discussion, and if necessary, a vote. In this context a reviewer is anyone leaving comments on the PR including contributors, committer and PMC members.
+
+There are several exceptions to this process:
+
+* Large changes and functional changes to a specification must go through the [Iceberg improvement proposal](#apache-iceberg-improvement-proposals) before any code can be merged.
+* Changes to files under the `format` directory and `open-api/rest-catalog*` are considered specification changes. Unless already covered under an Iceberg improvement proposal, specification changes require their own vote (e.g. bug fixes or specification clarifications). The vote follows the ASF [code modification](https://www.apache.org/foundation/voting.html#votes-on-code-modification) model and no lazy consensus modifier. Grammar, spelling and minor formatting fixes are exempted from this rule.
```

Comment:
Confirming my understanding here:

Functional spec changes go through improvement-process , which in the end require a code modification vote (3 +1 votes without lazy consensus modifier)

Any spec clarification still requires a code modification vote (3 +1 votes without lazy consensus modifier).  A current example we are working is #10793.

Grammar, spelling, and minor formatting still go through normal code review process.

Reply:
This is my understanding (I was trying to codify rules that have come across the mailing list recently).